lms/services/lis_result_sourcedid.py

```diff
@@ -54,6 +54,33 @@ def fetch_students_by_assignment(
             .all()
         )
 
+    def is_assignment_gradable(self, request):
+        """
+        Determine if the request has the parameters required to enable grading.
+
+        :param request: A pyramid request
+        :rtype: bool
+        """
+
+        return self.get_grading_parameters(request) is not None
```

Comment:
You could take this one step further and remove the `request` argument from `is_assignment_gradeable()` and `get_grading_parameters()`, since service factories receive the request from pyramid_services. The two methods could then become `@property`'s which would simplify the integration code (the views). Instead of:

```python
if lis_result_sourcedid_svc.is_assignment_gradable(self.request):
    ...
```

it would be just:

```python
if lis_result_sourcedid_svc.is_assignment_gradable:
    ...
```

It would also simplify the integration tests (view tests) as there's no need to test `lis_result_sourcedid_svc.is_assignment_gradable.assert_called_once_with(pyramid_request)` since it's no longer the view's responsibility to correctly pass the request in. Tests would just set `lis_result_sourcedid_svc.is_assignment_gradable` to `True` and to `False` (instead of setting `lis_result_sourcedid_svc.is_assignment_gradable.return_value` as they should currently) and assert that the views do the right thing for the `True` and `False` cases.